contents/ui/main.qml

```diff
@@ -175,18 +171,14 @@ QQC2.StackView {
     function pickImage(d) {
         if (d.data.length > 0) {
             var index = 0;
-            if (wallpaper.configuration.Sorting != "random") {
-                index = root.currentIndex;
-                if (index > 24) {
-                    root.currentPage += 1;
-                    root.currentIndex = 0;
-                    refreshTimer.restart();
-                    return;
-                }
-                root.currentIndex += 1;
-            } else {
-                index = Math.floor(Math.random() * d.data.length);
+            index = root.currentIndex;
+            if (index > 24) {
+                root.currentPage += 1;
+                root.currentIndex = 0;
+                refreshTimer.restart();
+                return;
             }
+            root.currentIndex += 1;
```

Comment:
I'm not sure, but since we're always going to get the wallpaper from page 1 starting from 0, this part needs to be within the if condition
```suggestion
            if (wallpaper.configuration.Sorting != "random") {
                index = root.currentIndex;
                if (index > 24) {
                    root.currentPage += 1;
                    root.currentIndex = 0;
                    refreshTimer.restart();
                    return;
                }
                root.currentIndex += 1;
            }
```